jupyterhub/spawners.py

```diff
@@ -70,7 +70,7 @@ def get_env(self):
             "CI_NAMESPACE": self.user_options.get("namespace", ""),
             "CI_PROJECT": self.user_options.get("project", ""),
             "CI_COMMIT_SHA": self.user_options.get("commit_sha", ""),
-            "GITLAB_URL": os.environ.get("GITLAB_URL", "http://gitlab.renku.build"),
+            "GITLAB_URL": os.environ["GITLAB_URL"],
```

Comment:
please use `os.environ.get` and handle the case of `None` - otherwise if it's not set you get an unhandled `KeyError` here.

Reply:
@rokroskar see the first comment in this PR. I thought we wanted things to crash if we do not specify the GIT_URL as an environment variable.

Reply:
Ah I missed that. Crashing with a KeyError is not the best way to handle a missing required config option imho - the app should fail already when `helm` tries to render the chart, but if deployed outside that and required config options are missing it should give a meaningful error and exit (in the cases where we can't provide a reasonable default).

Reply:
> Ah I missed that. Crashing with a KeyError is not the best way to handle a missing required config option imho
Not the best at all, but better than silently using `http://gitlab.renku.build` I would say.

> the app should fail already when helm tries to render the chart, but if deployed outside that and required config options are missing it should give a meaningful error and exit (in the cases where we can't provide a reasonable default).

Agree, but I think that's a bit outside the scope of this bugfix PR?

Reply:
https://github.com/SwissDataScienceCenter/renku-notebooks/issues/462

Reply:
Sure that's fine - I wasn't sure why this PR was touching that code, but I didn't think it should be done like that. :)

Reply:
Ok so I will roll back all the changes I made for `os.environ["GITLAB_URL"]` so that they are back to using `.get` with a hardcoded default. Then we will fix this up in the issue that @ableuler created.